docs/Plugin-BuildrootLock.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: default
+title: Plugin BuildrootLock
+---
+
+This plugin produces the `buildroot.lock` file in the result directory.  The
+lockfile describes the buildroot so it can be later reproduced by Mock (see the
+feature page, TODO).
+
+The BuildrootLock plugin is disabled by default, but automatically enabled with
+the `--calculate-build-dependencies` option.
+
+The file `buildroot.lock` is a JSON file.
+
+This plugin will not work correctly with `--offline` mode.
+
+## Format of the lockfile
+
+We currently do not provide a compatibility promise.  Only the very same version
+of Mock that produced the file is guaranteed to read and process it later (see
+the feature page, TODO).
+
+Example of such lockfile:
+
+    {
+        "version": "0",
+        "buildroot": {
+            "packages": [
+                {
+                    "license": "MIT",
+                    "name": "fedora-repos-rawhide",
+                    "version": "42",
+                    "release": "0.1",
+                    "arch": "noarch",
+                    "epoch": null,
+                    "nvra": "fedora-repos-rawhide-42-0.1.noarch",
+                    "sigmd5": "4378929dac9e51ed8470de5173ae664e",
+                    "signature": null,
+                    "url": "http://ftp.sh.cvut.cz/fedora/linux/development/rawhide/Everything/x86_64/os/Packages/f/fedora-repos-rawhide-42-0.1.noarch.rpm"
+                },
+                ...
+            ]
+        },
+        "config": {
```

Comment:
Do we really want this as array? I would probably put it here as multiline string.

Reply:
s/array/dict/?  Do you suggest having "dict" encoded as string inside the upper level dict?

Reply:
Ignore this. I just learn that multiline values are not supported in JSON.